bibliopixel/remote/server.py

```diff
@@ -0,0 +1,93 @@
+from flask import Flask, jsonify, send_from_directory
+from werkzeug.serving import run_simple
+import threading
+import queue
+import os
+import logging
+from .. import log
+
+# I want to keep this to supress the "Running on" message
+# But then it also swallows 404 erros and such in the console :(
+# werk_log = logging.getLogger('werkzeug')
+# werk_log.setLevel(logging.ERROR)
+
+
+def fail(msg='Error'):
+    return {
+        'status': False,
+        'msg': msg,
+        'data': None
+    }
+
+
+def success(data=None, msg='OK'):
+    return {
+        'status': True,
+        'msg': msg,
+        'data': data
+    }
+
+
+class RemoteServer():
+    def __init__(self, q_send, q_recv):
+        self.__server_thread = None
+        self.__recv_thread = None
+        cdir = os.path.dirname(os.path.realpath(__file__))
+        static_dir = os.path.abspath(os.path.join(cdir, '../../ui/web_remote'))
+        print(static_dir)
+        self.app = Flask('BP Remote', static_folder=static_dir)
+        self._set_routes()
+        self.q_send = q_send
+        self.q_recv = q_recv
+
+    def _set_routes(self):
+        self.app.route('/')(self.index)
+        self.app.route('/<path:path>')(self.static_files)
+        self.app.route('/run_animation/<string:animation>')(self.run_animation)
+        self.app.route('/stop')(self.stop_animation)
+        self.app.route('/api/<string:request>')(self.api)
+        self.app.route('/api/<string:request>/<data>')(self.api)
+
+    def index(self):
+        return self.app.send_static_file('index.html')
+
+    def static_files(self, path):
+        return self.app.send_static_file(path)
+
+    def run_animation(self, animation):
+        resp = self.api('run_animation', data=animation)
```

Comment:
resp = response
Was trying to keep the lines short. I generally prefer breaking things up into lines if I can.